tests/test_noema_truncated_completion_contract.py

```
@@ -0,0 +1,283 @@
+                    "path": "src/example.py",
+                    "line": 7,
+                    "side": "RIGHT",
+                    "analysis": ["x" * noema.NOEMA_MAX_VERDICT_TEXT_CHARS],
+                }
+            ],
+        }
+    )
+    opener = _Opener([_envelope(malformed, "stop"), _envelope(malformed, "stop")])
+    _configure(monkeypatch, opener)
+
+    with pytest.raises(RuntimeError, match="reviewed_lines.analysis must be a string"):
```

Ruff (0.16.3)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)